node_modules/gh-orgunit/lib/internal/dao.js

```diff
@@ -415,3 +416,377 @@ var getOrgUnitCalendar = module.exports.getOrgUnitCalendar = function(orgUnit, s
+ */
+var exportOrgUnit = module.exports.exportOrgUnit = function(orgUnit, format, callback) {
+    // Get all the organisational units under the given organisational unit
+    _getChildOrgUnits([orgUnit.id], function(err, orgUnitIds) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Get the event series and events under the organisational units
+        var options = {
+            'where': {
+                'id': orgUnitIds
+            },
+            'include': [
+                // Series that are associated to the organisational unit
+                {'model': DB.Serie, 'include': [
+                    {'model': DB.Event, 'include': [
+                        {'model': DB.User, 'as': 'Organisers'}
+                    ]}
+                ]}
+
+                // TODO: Events that are directly associated to the organisational unit
+            ]
+        };
+        DB.OrgUnit.findAll(options).complete(function(err, orgUnits) {
+            if (err) {
+                log().error({
+                    'err': err,
+                    'id': orgUnit.id
+                }, 'Unable to export an organisational unit');
+                return callback({'code': 500, 'msg': 'Unable to export an organisational unit'});
+            }
+
+            if (format === 'csv') {
+                return _exportOrgUnitCSV(orgUnit.id, orgUnits, callback);
+            } else {
+                return _exportOrgUnitJSON(orgUnit.id, orgUnits, callback);
+            }
+        });
+    });
+};
+
+/**
+ * Find all the organisational units under a set of organisational units. This function will
+ * perform a breadth-first search through the organisational unit structure generating in an SQL
+ * query per level. This means that if the maximum depth of one of the trees is 4, 4 SQL queries will be executed
+ *
+ * @param  {Number[]}   orgUnitIds                  The ids of the organisational units to expand
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error object, if any
+ * @param  {Number[]}   callback.allOrgUnitIds      The ids of the organisational units that are part of the trees (including the original root ids)
+ * @api private
+ */
+var _getChildOrgUnits = function(orgUnitIds, callback, _allOrgUnitIds) {
+    _allOrgUnitIds = _allOrgUnitIds || orgUnitIds;
+
+    // Get the children of the specified organisational units
+    _getDirectChildOrgUnits(orgUnitIds, function(err, childOrgUnitIds) {
+        if (err) {
+            return callback(err);
+        }
+
+        // If no further children could be produced, we return
+        if (_.isEmpty(childOrgUnitIds)) {
+            return callback(null, _allOrgUnitIds);
+        }
+
+        // Add the children to the set of known organisational unit ids and get the children's children
+        _allOrgUnitIds = _.union(_allOrgUnitIds, childOrgUnitIds);
```

Comment:
Is there a scenario in which an orgunit can be in multiple locations in the hierarchy? If so, we'd need to remove duplicates from `_allOrgUnitIds` and would need to remove orgunits we've already seen from `childOrgUnitIds`